### PR TITLE
cppdap: update to 1.65

### PR DESCRIPTION
**Pull request by jcaselman-keysight**
* Updates cppdap to [ver 1.65 ](https://github.com/google/cppdap/commit/37c744c294cbb24943a3913bbb67860d1609008c)
  * Switch to git commit source since the project does not generate github release archives

### CODE REVIEW

**Comment by ognevny** [comment]
please, run `updpkgsums`. starting with pacman 6.1.0, checksums for repositories are supported

File: mingw-w64-cppdap/PKGBUILD [hunk 1]
```diff
@@ -1,22 +1,25 @@
 # Contributor: Mehdi Chinoune <mehdi.chinoune@hotmail.com>
+# Contributor: Justin Caselman <justin.caselman@keysight.com>
 
 _realname=cppdap
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-_pkgver=1.58.0-a
+_pkgver=1.65
 pkgver=${_pkgver/-/}
+_commit="37c744c294cbb24943a3913bbb67860d1609008c"
```
Comment (lazka) [inline mingw-w64-cppdap/PKGBUILD:9]:
```suggestion
# tags are missing, see https://github.com/google/cppdap/issues/132
_commit="37c744c294cbb24943a3913bbb67860d1609008c"
```

**Comment by MehdiChinoune** [comment]
I think It's possible now to build shared library.

**Comment by jcaselman-keysight** [comment]
> please, run `updpkgsums`. starting with pacman 6.1.0, checksums for repositories are supported

done

> I think It's possible now to build shared library.

I'll convert to draft while I look into a shared lib

**Comment by MehdiChinoune** [comment]
Please add `depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")` to satisfy runtime dependencies.

**Comment by jcaselman-keysight** [comment]
> Please add `depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")` to satisfy runtime dependencies.

@MehdiChinoune done.  Is this a tribal knowledge thing or how was this surfaced as being necessary?

**Comment by MehdiChinoune** [comment]
Take a look at 'runtime-dependencies' in CI.
They were not required because It was built as a static library.

**Comment by jcaselman-keysight** [comment]
> Take a look at 'runtime-dependencies' in CI. They were not required because It was built as a static library.

I see,

https://github.com/msys2/MINGW-packages/actions/runs/9982129129/job/27587276214#step:11:728
vs
https://github.com/msys2/MINGW-packages/actions/runs/9982613429/job/27588683729?pr=21437#step:11:750

Thank you 😄

**Comment by MehdiChinoune** [comment]
I have tried to rebuild cmake against it but It fails because cppdap doesn't export required symbols.
Reverting to static or adding `LDFLAGS+=" -Wl,--export-all-symbols"` fixes the issue. I prefer the second solution.
```
  LDFLAGS+=" -Wl,--export-all-symbols" \
  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
```

**Pull request merged**